### PR TITLE
Support for scalar arguments in Np.ascontiguousarray 

### PR DESCRIPTION
**Pull request by DhruvPatel01**
## Reference an existing issue

#7353 

## Details

I have added a code to overload scalar arguments to `np.ascontiguousarray` and `np.asfortranarray`. Existing implementations use low level API `glue_lowering`, but I couldn't figure out how to modify the code to handle scalars. Hopefully mixing high level API (`overload`) and low level API is fine.

### CODE REVIEW

**Comment by DhruvPatel01** [comment]
I am not sure why build is failing in Python 3.10 Linux. The failed cases seem unrelated to this commit.

**Comment by njriasan** [comment]
Hi @DhruvPatel01. Regarding I'll defer to maintainers for a more complete answer to `Hopefully mixing high level API (overload) and low level API is fine`, but I think it should be okay in this case because the type checks will never overlap.

For my testing this implementation/review this implementation seems correct but, it would be great if you could add a test to verify your implementation is working as expected.

It also seems to me that the build is failing in Python 3.10 Linux due to an unrelated error.

Just as a note I think many of the maintainers are still on a holiday break this week, so there may be a delay in getting more detailed feedback.

File: numba/np/arrayobj.py
Comment (njriasan) [inline numba/np/arrayobj.py:4352]: Personally I think it would be helpful if you could add a docstring here explaining this handles the scalar implementation and that the array cases are handled with the lower level glue_* APIs. Someone reading through this who is use to searching for overloads may be surprised when this is all they find.

File: numba/np/arrayobj.py
Comment (njriasan) [inline numba/np/arrayobj.py:4334]: I'm not too familiar with the Fortran array layout (I never use it), but I suspect reusing the same function here is not completely correct because although it shouldn't be meaningful, the types won't accurately reflect the desired layout.

In particular based on existing code, this will always give the `layout='C'`, which won't be correct for the Fortran function. While this probably isn't meaningful in this case, I'm not sure what impact it might have when mixed with other APIs.

I'm not sure if this can be done with the higher level APIs, but I'll defer to people who are more knowledgable (perhaps @stuartarchibald).

**Review by njriasan — COMMENTED** [review]
Thanks for submitting this. I'm not a maintainer, but I work with Numba a decent amount and figured I'd try help with this PR. The biggest issue is that in Numba, an array layout is part of type (i.e. C style or Fortran style) and reusing this higher level API will cause the fortran style to have an incorrect type.

I think this requires the lower level API at this time, but I'll let @stuartarchibald give the final word about that. If you need some help modifying your solution to integrate with the existing solution and include the correct layout I'm happy to help guide you through the code.

File: numba/np/arrayobj.py
Comment (DhruvPatel01) [reply numba/np/arrayobj.py:4352]: Agree, will add docstrings for it.

File: numba/np/arrayobj.py
Comment (DhruvPatel01) [reply numba/np/arrayobj.py:4334]: For single element arrays, I think they should be the same. I printed `np.ascontiguousarray(1).flags`. Both  `C_CONTIGUOUS ` and `F_CONTIGUOUS` were `True`. Same flags apart from `OWNDATA` flag were returned by this implementation. This also applies for `asfortranarray`.

**Comment by stuartarchibald** [comment]
> I am not sure why build is failing in Python 3.10 Linux. The failed cases seem unrelated to this commit.

https://github.com/numba/numba/pull/7702 should hopefully fix.

File: numba/np/arrayobj.py
Comment (DhruvPatel01) [reply numba/np/arrayobj.py]: Added the docstrings.

**Review by esc — APPROVED** [review]
Thank you for the patch!

**Comment by stuartarchibald** [comment]
Thanks for the patch @DhruvPatel01, thanks for reviewing @njriasan @esc.

**Pull request merged**
